Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs

```diff
@@ -224,254 +355,190 @@ public void ShowAsWindow(eWindowShowStyle windowStyle = eWindowShowStyle.Free)
             //if we come from ActionEditPage we need to give the user to update only Action selector + show him some message how to do it
             if (mAction != null)
             {
-                ActionsTab.Header = "Action Locator Selection";
-                SelectedControlDetailsExpander.IsExpanded = true;
-                ActionsTab.IsSelected = true;
-            } 
+                //ActionsTab.Header = "Action Locator Selection";
+                //SelectedControlDetailsExpander.IsExpanded = true;
+                //ActionsTab.IsSelected = true;
+            }
             else
             {
-                StepstoUpdateActionRow.Height = new GridLength(0);
+                //StepstoUpdateActionRow.Height = new GridLength(0);
             }
-            GingerCore.General.LoadGenericWindow(ref _GenWin, null, windowStyle, Title, this, closeEventHandler: CloseWindow);            
+            GingerCore.General.LoadGenericWindow(ref _GenWin, null, windowStyle, Title, this, closeEventHandler: CloseWindow);
         }
 
         private async void CloseWindow(object sender, EventArgs e)
         {
             //stop Live Spy
-            if (LiveSpyButton.IsChecked == true)
-                if (dispatcherTimer != null)
-                    dispatcherTimer.IsEnabled = false;
+            //if (LiveSpyButton.IsChecked == true)
+            //    if (dispatcherTimer != null)
+            //        dispatcherTimer.IsEnabled = false;
 
             //stop Recording
-            if (RecordingButton.IsChecked == true)
-                StopRecording();
+            //if (RecordingButton.IsChecked == true)
+            //    StopRecording();
 
-            if(WindowControlsTreeView.IsSearchRunning())
+            if (xWindowControlsTreeView.IsSearchRunning())
             {
-                await WindowControlsTreeView.CancelSearchAsync();
+                await xWindowControlsTreeView.CancelSearchAsync();
             }
 
             _GenWin.Close();
         }
 
         void InitTree(ITreeViewItem RootItem)
         {
-            WindowControlsTreeView.Tree.ClearTreeItems();
+            xWindowControlsTreeView.Tree.ClearTreeItems();
             mRootItem = RootItem;
 
-            mTreeRootItem = WindowControlsTreeView.Tree.AddItem(RootItem);
-            mTreeRootItem.IsExpanded = false;            
+            mTreeRootItem = xWindowControlsTreeView.Tree.AddItem(RootItem);
+            mTreeRootItem.IsExpanded = false;
         }
-        
-        private void RefreshTreeControls()
+
+        private async Task RefreshPageSrcContent()
         {
-            WindowControlsTreeView.Tree.ClearTreeItems();
-            if (WindowsComboBox.SelectedValue != null && mRootItem != null)
+            //xHTMLTree.Items.Add(new PageSrcParser(((SeleniumDriver)mContext.Agent.Driver).GetPageHTML())
+            if (mWindowExplorerDriver is GenericAppiumDriver)
```

Comment:
why specific check for AppiumDriver? need to use interface

Reply:
specific check as for Web, it's HTMLViewer while for Mobile, it's XML. I'm planning to make it in a more generic way by having a single User Control (say, PageSourceViewer by combining XMLViewer & HTMLViewer in a single) to handle any document sent and show the page source in Explorer. It'll take time so keeping as it is for now, or shall i shift in the same PR, today ?